src/apify/storage_clients/_apify/_key_value_store_client.py

```diff
@@ -0,0 +1,238 @@
+
+    @override
+    async def delete_value(self, key: str) -> None:
+        async with self._lock:
+            await self._api_client.delete_record(key=key)
+            await self._update_metadata()
+
+    @override
+    async def iterate_keys(
+        self,
+        *,
+        exclusive_start_key: str | None = None,
+        limit: int | None = None,
+    ) -> AsyncIterator[KeyValueStoreRecordMetadata]:
+        count = 0
+
+        while True:
+            response = await self._api_client.list_keys(exclusive_start_key=exclusive_start_key)
+            list_key_page = KeyValueStoreListKeysPage.model_validate(response)
+
+            for item in list_key_page.items:
+                # Convert KeyValueStoreKeyInfo to KeyValueStoreRecordMetadata
+                record_metadata = KeyValueStoreRecordMetadata(
+                    key=item.key,
+                    size=item.size,
+                    content_type='application/octet-stream',  # Content type not available from list_keys
+                )
+                yield record_metadata
+                count += 1
+
+                # If we've reached the limit, stop yielding
+                if limit and count >= limit:
+                    break
+
+            # If we've reached the limit or there are no more pages, exit the loop
+            if (limit and count >= limit) or not list_key_page.is_truncated:
+                break
+
+            exclusive_start_key = list_key_page.next_exclusive_start_key
+
+        await self._update_metadata()
+
+    @override
+    async def record_exists(self, key: str) -> bool:
+        return await self._api_client.record_exists(key=key)
+
+    async def get_public_url(self, key: str) -> str:
+        """Get a URL for the given key that may be used to publicly access the value in the remote key-value store.
+
+        Args:
+            key: The key for which the URL should be generated.
+
+        Returns:
+            A public URL that can be used to access the value of the given key in the KVS.
+        """
+        if self._api_client.resource_id is None:
+            raise ValueError('resource_id cannot be None when generating a public URL')
+
+        public_url = (
+            URL(self._api_public_base_url) / 'v2' / 'key-value-stores' / self._api_client.resource_id / 'records' / key
+        )
+        metadata = await self.get_metadata()
+
+        if metadata.model_extra is not None:
+            url_signing_secret_key = metadata.model_extra.get('urlSigningSecretKey')
+            if url_signing_secret_key is not None:
+                public_url = public_url.with_query(signature=create_hmac_signature(url_signing_secret_key, key))
+
+        return str(public_url)
+
+    async def _update_metadata(self) -> None:
+        """Update the key-value store metadata with current information."""
+        metadata = await self._api_client.get()
+        self._metadata = KeyValueStoreMetadata.model_validate(metadata)
```

Comment:
Is this ever used? I recall discussing not storing the metadata, but I'm not sure what the conclusion was.